src/main/scala/cognite/spark/v1/FileContentRelation.scala

```diff
@@ -0,0 +1,108 @@
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.sources.{BaseRelation, Filter, PrunedFilteredScan, TableScan}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, Row, SQLContext, SparkSession}
+import org.apache.spark.{Partition, TaskContext}
+import sttp.client3.{HttpURLConnectionBackend, UriContext, basicRequest}
+
+import java.net.URL
+import scala.io.Source
+
+trait WithSizeLimit {
+  val sizeLimit: Long
+}
+
+class FileContentRelation(config: RelationConfig, fileExternalId: String)(
+    override val sqlContext: SQLContext)
+    extends BaseRelation
+    with TableScan
+    with PrunedFilteredScan
+    with Serializable
+    with WithSizeLimit {
+
+  override val sizeLimit: Long = 5 * FileUtils.ONE_GB
+
+  @transient lazy val client: GenericClient[IO] =
+    CdpConnector.clientFromConfig(config)
+
+  override def schema: StructType =
+    createDataFrame(sqlContext.sparkSession).schema
+
+  def createDataFrame(sparkSession: SparkSession): DataFrame = {
+    val rowsRdd: RDD[String] = new RDD[String](sparkSession.sparkContext, Nil) with Serializable {
+
+      import cognite.spark.v1.CdpConnector.ioRuntime
+
+      val maxParallelism = 1
+      override def compute(split: Partition, context: TaskContext): Iterator[String] = {
+
+        val validUrl = for {
+          mimeType <- client.files.retrieveByExternalId(fileExternalId).map(_.mimeType)
+          downloadLink <- client.files
+            .downloadLink(FileDownloadExternalId(fileExternalId))
+            .map(_.downloadUrl)
+          isFileWithinLimits <- isFileWithinLimits(downloadLink)
+        } yield {
+          if (!isFileWithinLimits)
+            throw new CdfSparkException("File size above 5Gb limit")
+          if (mimeType.isEmpty || !mimeType.contains("application/jsonlines"))
+            throw new CdfSparkException("Wrong mimetype. Expects application/jsonlines")
+          else
+            downloadLink
+        }
+
+        StreamIterator(
+          Stream
+            .eval(validUrl)
+            .flatMap(readUrlContent),
+          maxParallelism,
+          None
+        )
+
+      }
+
+      override protected def getPartitions: Array[Partition] =
+        Array(CdfPartition(0))
+    }
+    import sparkSession.implicits._
+    sparkSession.read.json(rowsRdd.toDS())
+  }
+
+  private def readUrlContent(url: String): Stream[IO, String] =
+    Stream
+      .bracket(IO(new URL(url).openStream())) { inStream =>
+        IO(inStream.close()) // Ensure the InputStream is closed properly
+      }
+      .flatMap { inStream =>
+        val source = Source.fromInputStream(inStream)
+        Stream.fromIterator[IO](source.getLines(), chunkSize = 4096)
+      }
+
+  private def isFileWithinLimits(downloadUrl: String): IO[Boolean] = {
+    val backend = Resource.make(IO(HttpURLConnectionBackend()))(b => IO(b.close()))
+    val request = basicRequest.head(uri"$downloadUrl")
```

Comment:
may consider having such method in the sdk :thinking: 
Also I guess everywhere those links are multi-use?

Reply:
actually looks like there's
```
  def download(item: FileDownload, out: java.io.OutputStream): F[Unit] = {
```
but would be good to check how good it is wrt streaming and early closing

Reply:
the sdk method above is for download, no method for HEAD request to get size yet

Reply:
I can't know for sure if they are multi use on all clusters unfortunately. It should be but we might encounter a bad surprise later.... head request is the only way to make this work, otherwize we can't know the size of the file without downloading it first.

The reason I didn't add it to the SDK quite yet is because it doesn't require credentials or anything here, so it's just a very basic request.

Reply:
Also when it comes to the download link, I'm pretty sure it didn't fit my usecase unfortunately so I didn't use it. I think my code changed since so I'll check if that still holds true.

Reply:
There's already full upload&download support there https://github.com/cognitedata/cognite-sdk-scala/blob/master/src/main/scala/com/cognite/sdk/scala/v1/resources/files.scala#L160 so most of the plumbing should be in place and it may make PR tiny bit slimmer on datasource side. Won't be a blocker, but potentially may actually speed up review by splitting into 2 PRs and reusing some of SDK :slightly_smiling_face:

Reply:
I'm pretty ignorant about how the different backend interact and what they do I'll read up on it